client/me/purchases/manage-purchase/plan-details/index.jsx

```diff
@@ -98,9 +109,12 @@ class PurchasePlanDetails extends Component {
 }
 
 // hasLoadedSites & hasLoadedUserPurchasesFromServer are used in isDataLoading
-export default connect( ( state, props ) => ( {
-	hasLoadedSites: ! isRequestingSites( state ),
-	hasLoadedUserPurchasesFromServer: hasLoadedUserPurchasesFromServer( state ),
-	purchase: getByPurchaseId( state, props.purchaseId ),
-	pluginList: props.selectedSite ? getPluginsForSite( state, props.selectedSite.ID ) : [],
-} ) )( localize( PurchasePlanDetails ) );
+export default connect( ( state, props ) => {
+	const purchase = getByPurchaseId( state, props.purchaseId );
+	return {
+		hasLoadedSites: ! isRequestingSites( state ),
+		hasLoadedUserPurchasesFromServer: hasLoadedUserPurchasesFromServer( state ),
+		purchase,
+		pluginList: purchase ? getPluginsForSite( state, purchase.siteId ) : [],
```

Comment:
You could consider passing `[purchase]siteId` as a prop here since you're using it inside the component.

Reply:
You mean extracting `siteId` from `purchase` and passing it as a prop in `connect`?

Reply:
Yep!

Reply:
👍 450c506793ac5cbf8004309059b5dacc9953a792